src/dbnode/storage/stats/query_stats.go

```diff
@@ -55,33 +57,47 @@ type QueryStatsOptions struct {
 	// MaxDocs limits how many recently queried max
 	// documents are allowed before queries are abandoned.
 	MaxDocs int64
-	// Lookback specifies the lookback period over which stats are aggregated.
-	Lookback time.Duration
+	// MaxDocsLookback specifies the lookback period over which max docs limit is enforced.
+	MaxDocsLookback time.Duration
+	// MaxBytesRead limits how many recently queried bytes
+	// read from disk are allowed before queries are abandoned.
+	MaxBytesRead int64
+	// MaxBytesReadLookback specifies the lookback period over which max bytes read limit is enforced.
+	MaxBytesReadLookback time.Duration
 }
 
 // QueryStatsValues stores values of query stats.
 type QueryStatsValues struct {
-	RecentDocs int64
-	NewDocs    int64
+	// DocsMatched is a sum of all docs matched by recent queries.
+	DocsMatched QueryStatsValue
+	// BytesRead is the sum of bytes read from disk by recent queries.
+	BytesRead QueryStatsValue
 }
 
-var zeros = QueryStatsValues{
-	RecentDocs: 0,
-	NewDocs:    0,
+// QueryStatsValue stores values of a particular query
+// stat being tracked within a lookback period.
+type QueryStatsValue struct {
+	// Recent is the total count having occurred in the most recent lookback period.
+	Recent int64
+	// New is the latest incremental count to have occurred.
+	New int64
+	// Reset marks whether a lookback period has elapsed and therefore the count reset to zero.
```

Comment:
you need this in order to distinguish the a default value of 0 vs a reset value of 0?

Reply:
It is actually a bit more confusing actually because of how we want to alter how we emit this metric. The issue is that - in the case where we are doing a "reset" (i.e. a lookback period has elapsed so we want to start `recent` again at zero), the val we want to set the gauge metric to is the peak/latest val before that reset (currently we don't do this which makes the metric confusing to read since it jumps back to zero every time instead of having a smoother trend). To do this we need to also have the latest recent val passed (and as you said, if we infer a reset of a `Recent` val of zero then we don't actually have that latest peak val to emit).

An alternative would be to keep the previous recent val in the `tracker` itself, but then 

An al

Reply:
interesting. i think ideally there would be some kind of `TickerGauge` that would encapsulate the reset logic based on a configured ticker. not sure how generally useful that would be. I've seen similar impls for a `PeakGauge`, but they are tied to the collection interval, which you don't want. just something to keep in mind in case we see this pattern again.